### PR TITLE
feat(cli:babel-loader): cli2.x add pnpm support

### PR DESCRIPTION
**Pull request by zhangyuhan2016**
#10483 @JakeLaoyu

### CODE REVIEW

File: packages/vant-cli/src/config/webpack.base.ts [hunk 1]
```diff
@@ -84,7 +84,7 @@ export const baseConfig: WebpackConfig = {
       },
       {
         test: /\.(js|ts|jsx|tsx)$/,
-        exclude: /node_modules\/(?!(@vant\/cli))/,
+        exclude: /node_modules(\/.pnpm){?}\/(?!(@vant\+cli))/,
```
Comment (chenjiahan) [inline packages/vant-cli/src/config/webpack.base.ts:87]: image

`.pnpm` should be optional:

`/node_modules(\/.pnpm)?\/(?!(@vant\+cli))/`

image

Comment (zhangyuhan2016) [reply packages/vant-cli/src/config/webpack.base.ts]: hi~ [chenjiahan](https://github.com/chenjiahan), 

after inspection,
`/node_modules(\/\.pnpm)?\/(?!(@vant\/cli))/` cannot be used normally under pnpm,
so use `/^node_modules(\/\.pnpm)?\/(?!(@vant\/cli))/`can work normally.

thank you for taking the time to check.

image

**Review by chenjiahan — APPROVED** [review]

**Comment by chenjiahan** [comment]
👍🏻

**Comment by chenjiahan** [comment]
Released `@vant/cli@2.11.6`

**Pull request merged**